### ISSUE TITLE
Thoughts on children VDRs inside parent VDR (handling recursion)

### ISSUE THREAD

**Issue by alxpez** [mauricius/vue-draggable-resizable#28]
I stumbled upon the case-scenario in which a VDR may have other VDRs as children. Having the prop `:parent=true` maintains the children within the parent boundaries (as expected GOOD!), but the activation mechanism (meaning whether or not a VDR is `active`) occurs in all children VDRs (included the parent VDR). In this case, when trying to drag a child, the parent would move as well (once the child has reached the parent boundaries).

Probably not the best explanation (hard to do it in words... I'll try to post a visual example), but I hope my point goes through. I will be looking into [vdr](https://github.com/mauricius/vue-draggable-resizable/blob/master/src/components/vue-draggable-resizable.vue) code to find possible solutions. However, this feature might not be within your development plans.

Any thoughts on this? Cheers!

**Comment by mauricius** [comment]
Your point is clear, but as you said it is not an essential feature at this point. However if you are able to make it work without twisting the component (I believe that the issue is related to the propagation of mouse events), I might accept a PR.

Cheers!

**Comment by alxpez** [comment]
> I believe that the issue is related to the propagation of mouse events

That's right!

I've been able to create _well behaved_ children (and down to any depth... grandchildren, etc). 

The implementation was actually very simple and it actually lead me to solve something I had noticed before, which is that when the container (parent) of a VDR changes its dimension (after being rendered), the VDR element is unaware of it and will move out of bounds (or have invisible walls). 

With this changes, even if the parent changes size after the element is `mounted`, the VDR knows what the limits are.

![vdr-recursion](https://user-images.githubusercontent.com/5973189/30691742-ce164aae-9f1c-11e7-9dba-9b3a260d2f98.gif)
(the red box sudden jump happens when it gets clicked, re-calculates its parent limits and gets inside its bounds, there are other options, but that one seemed the simplest one)

I'll be pushing this commit and opening a PR, let me know!
